### DISCUSSION TITLE
Query Invalidation -> Requery -> Fetching state not updated ?

### DISCUSSION THREAD

**Discussion by cevr** [urql-graphql/urql#2009] [Q&A] [answered]
Is it intended that the `result.fetching` state is not updated when a query is invalidated?

After invalidating a query, I see that the query is refetched but the fetching state does not change.

Is this intended?

**Comment by JoviDeCroock** [comment, marked as answer]
Hmm, in what case specifically? Is this a document-based cache automatic invalidation or a graphcache updater calling `cache.invalidate` we'll need a bit more information regarding the problem here.

You can also check whether `result.stale` is true 😅 this often indicates that a background refetch is happening

**Reply by cevr** [reply]
Hi! It's a graphcache updater that is calling `cache.invalidate`. Yeah I've included result.stale as well in my conditionals. Is this the intended behaviour?

**Reply by JoviDeCroock** [reply]
It's intended that stale goes to `true` yes as you can see [here](https://codesandbox.io/s/confident-engelbart-pxnqg?file=/src/index.js) when you click a todo, we invalidate all todos on Query and we notify the UI that the data currently displayed is `stale` and it refetches in the background
